### ISSUE TITLE
`get_range` with `heed` does not work.

### ISSUE THREAD

**Issue by Boog900** [Cuprate/cuprate#348]
## Bug
`heed`'s  `get_range` function returns incorrect results when a different ordering is used with the DB/table.

## Steps to reproduce
Add this:
```rust
        for (v, vv) in table.get_range(100..249).unwrap().zip(100..249) {
            assert_eq!(v.unwrap(), vv);
        }
```

After this:
https://github.com/Cuprate/cuprate/blob/f3c1a5c2aa4629bf69b75268de21fc9112f09405/storage/database/src/backend/tests.rs#L377-L386

---

I think this is because heed's range cursor compares raw bytes here: https://docs.rs/heed/latest/src/heed/iterator/range.rs.html#416

**Comment by Boog900** [comment]
Yeah slightly embarrassing but I went too aggressive on trying to simplify the test case, updated the issue with the new test case that actually fails due to this issue.

**Comment by Boog900** [comment]
https://github.com/meilisearch/heed/pull/289

**Comment by SyntheticBird45** [comment]
`heed` has been replaced with Tapes and Fjall
